src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,151 @@
+package duke;
+
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ErrorCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.HelloCommand;
+import duke.command.ListCommand;
+import duke.command.SnoozeCommand;
+import duke.command.TodoCommand;
+import duke.command.UndoCommand;
+
+/**
+ * Parser object that parses all the input from user to commands understood by Duke
+ */
+public class Parser {
+    private static DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/uuuu HHmm");
+    private static String[] parsed;
+
+    private static int parseTaskNo(String parse) {
+        int taskNo = Integer.parseInt(parse) - 1;
+        return taskNo;
+    }
+
+    private static String parseTodoTask() throws IllegalArgumentException {
+        if (parsed.length < 2 || parsed[1].equals("")) {
+            throw new IllegalArgumentException();
+        }
+        return parsed[1];
+    }
+
+    private static String[] parseEventTask(String command) throws
+            IllegalArgumentException,
+            ArrayIndexOutOfBoundsException,
+            StringIndexOutOfBoundsException {
+        String[] eventDetails = new String[2];
+        String desc = command.split(" /at ")[0].substring(6);
```

Comment:
According to the code quality, you should minimise the usage of magic numbers like `6` in this case. Perhaps you can declare a constant in this class and change `6` to something like `START_INDEX_EVENT = EventCommand.COMMAND.length() + 1`? same for the other parse methods